app/expirynotify.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/utils"
+)
+
+const (
+	OneHourMillis = 60 * 60 * 1000
+)
+
+// NotifySessionsExpired is called periodically from the job server to notify any mobile sessions that have expired.
+func (a *App) NotifySessionsExpired() *model.AppError {
+	if *a.Config().EmailSettings.SendPushNotifications {
+		pushServer := *a.Config().EmailSettings.PushNotificationServer
+		if license := a.License(); pushServer == model.MHPNS && (license == nil || !*license.Features.MHPNS) {
+			mlog.Warn("Push notifications are disabled. Go to System Console > Notifications > Mobile Push to enable them.")
+			return nil
+		}
+	}
+
+	// Get all mobile sessions that expired within the last hour.
+	sessions, err := a.srv.Store.Session().GetSessionsExpired(OneHourMillis, true, true)
+	if err != nil {
+		return err
+	}
+
+	msg := &model.PushNotification{
+		Category: model.CATEGORY_CAN_REPLY,
```

Comment:
you cannot really reply to this notification

Reply:
Good point. What would the correct category be?  Blank?  CAN_REPLY is the only one currently used.

Reply:
@wiggin77 yeah, just leave it blank or don't set it

Reply:
Fixed